fields/types/key/KeyType.js

```diff
@@ -18,6 +18,8 @@ util.inherits(key, FieldType);
 
 /* Inherit from TextType prototype */
 key.prototype.addFilterToQuery = TextType.prototype.addFilterToQuery;
+key.prototype.validateInput = TextType.prototype.validateInput;
+key.prototype.validateRequiredInput = TextType.prototype.validateRequiredInput;
```

Comment:
This needs to be custom, but we can deal with that in a separate PR

The validation should run it through the slug value (which will remove certain characters) and check that it still has length, or we could persist an empty string in a `required` field